pkg/controller/persistentvolume/persistentvolume_claim_binder_controller.go

```diff
@@ -305,17 +326,41 @@ func syncClaim(volumeIndex *persistentVolumeOrderedIndex, binderClient binderCli
 			return fmt.Errorf("Unexpected error getting persistent volume: %v\n", err)
 		}
 
+		// TODO:  Can this be removed? The ref will not be nil for volumes bound by this controller.
 		if volume.Spec.ClaimRef == nil {
 			glog.V(5).Infof("Rebuilding bind on pv.Spec.ClaimRef\n")
 			claimRef, err := api.GetReference(claim)
 			if err != nil {
 				return fmt.Errorf("Unexpected error getting claim reference: %v\n", err)
 			}
-			volume.Spec.ClaimRef = claimRef
-			_, err = binderClient.UpdatePersistentVolume(volume)
+			clone, err := conversion.NewCloner().DeepCopy(volume)
 			if err != nil {
+				return fmt.Errorf("Error cloning pv: %v", err)
+			}
+			volumeClone, ok := clone.(*api.PersistentVolume)
+			if !ok {
+				return fmt.Errorf("Unexpected pv cast error : %v\n", volumeClone)
+			}
+			volumeClone.Spec.ClaimRef = claimRef
+			if updatedVolume, err := binderClient.UpdatePersistentVolume(volumeClone); err != nil {
 				return fmt.Errorf("Unexpected error saving PersistentVolume.Status: %+v", err)
+			} else {
+				volumeIndex.Update(updatedVolume)
+			}
+		}
+
+		// mismatched claims and bound volumes is an incorrect state.
+		// reset to Pending until api.ClaimError phase is introduced.
+		// the claim could become bound to another volume, which is appropriate.  This current volume
+		// does not belong to the claim.
+		if !claimAndVolumeMatch(claim, volume) {
```

Comment:
> Users can mutate the PVC and set the VolumeName, but Status is not persistable by normal users.
> 
> It was originally intended that users can claim/request specific volumes just as they can nodes (with pod nodeSelector). Volumes/Claims parallels Nodes/Pods. I'm no longer convinced that is correct or desirable for volumes.
> 
> Without this check, a user could create a claim, let it bind normally (to get into Bound state), and then change VolumeName. I believe we need an error state for this condition.

Why would I ever want to allow them to change a `Spec` **after** we're in the `bound` state?  Before, sure, but afterwards?